packages/mobile/src/app/selectors.ts

```diff
@@ -180,29 +182,48 @@ export const registrationStepsSelector = createSelector(
     biometryEnabledSelector,
     activeScreenSelector,
     recoveringFromStoreWipeSelector,
+    skipVerificationSelector,
   ],
-  (chooseRestoreAccount, biometryEnabled, activeScreen, recoveringFromStoreWipe) => {
+  (
+    chooseRestoreAccount,
+    biometryEnabled,
+    activeScreen,
+    recoveringFromStoreWipe,
+    skipVerification
+  ) => {
+    let steps
+    let totalSteps
+    let step
     if (recoveringFromStoreWipe) {
-      return {
-        step: storeWipeRecoverySteps[activeScreen as StoreWipeRecoveryScreens],
-        totalSteps: 3,
-      }
+      steps = storeWipeRecoverySteps
+      totalSteps = 3
+      step = storeWipeRecoverySteps[activeScreen as StoreWipeRecoveryScreens]
+    } else if (chooseRestoreAccount) {
+      steps = restoreAccountSteps
+      totalSteps = 5
+      step = restoreAccountSteps[activeScreen as RestoreAccountScreens]
+    } else {
+      steps = createAccountSteps
+      totalSteps = 4
+      step = createAccountSteps[activeScreen as CreateAccountScreens]
     }
 
-    if (chooseRestoreAccount) {
-      if (biometryEnabled) {
-        return { step: restoreAccountSteps[activeScreen as RestoreAccountScreens], totalSteps: 5 }
+    if (!biometryEnabled) {
+      if (Object.keys(steps).includes(Screens.EnableBiometry)) {
+        totalSteps--
+        step =
+          step > (steps as Record<Screens.EnableBiometry, number>)[Screens.EnableBiometry]
+            ? step - 1
+            : step
       }
-      // remove biometry screen from step
-      const step = restoreAccountSteps[activeScreen as RestoreAccountScreens]
-      return { step: step > 3 ? step - 1 : step, totalSteps: 4 }
+    }
+    if (skipVerification) {
+      totalSteps--
     }
 
-    if (biometryEnabled) {
-      return { step: createAccountSteps[activeScreen as CreateAccountScreens], totalSteps: 4 }
+    return {
+      step,
+      totalSteps,
     }
-    // remove biometry screen from step
-    const step = createAccountSteps[activeScreen as CreateAccountScreens]
-    return { step: step > 3 ? step - 1 : step, totalSteps: 3 }
```

Comment:
I noticed previously the code would return out of the if blocks when conditions were hit. Here it looks like here we are switching to modifying the `totalSteps`, `step` and `steps` across the if blocks as there are now multiple conditions; e.g. I could have !biometryEnabled and also be in the experiment group so `totalSteps`, `step` and `steps` so it needs to dynamically account for those two conditions; Is this correct? I don't doubt that this works, but it's a bit hard to follow.

Reply:
Yes thats correct, it needs to be able to account for the possibility of both the biometrics screen and the phone verification screens being skipped.

The code looks much more confusing in diff mode than it does in split diff or on its own.